### PR TITLE
Fix buffer overflow caught by valgrind

### PR DESCRIPTION
**Pull request by dankegel**
Fixes https://github.com/cinder/Cinder/issues/1554

### CODE REVIEW

**Comment by richardeakin** [comment]
Great, thanks for this - and the simplified toString() is nice. Also thanks for doing this against master branch.

File: test/unit/src/Base64Test.cpp [hunk 1]
```diff
@@ -25,8 +20,15 @@ TEST_CASE("Base64")
 		REQUIRE( toBase64( "any carnal pleasur" ) == "YW55IGNhcm5hbCBwbGVhc3Vy" );
 		REQUIRE( toBase64( "any carnal pleasu" ) == "YW55IGNhcm5hbCBwbGVhc3U=" );
 		REQUIRE( toBase64( "any carnal pleas" ) == "YW55IGNhcm5hbCBwbGVhcw==" );
+		REQUIRE( toBase64( "any" ) == "YW55" );
+		REQUIRE( toBase64( "an" ) == "YW4=" );
+		REQUIRE( toBase64( "a" ) == "YQ==" );
+		REQUIRE( toBase64( "" ) == "" );
+		REQUIRE( "" == toString( fromBase64("") ) );
```
Comment (richardeakin) [inline test/unit/src/Base64Test.cpp]: Style nit: spaces within function braces (see [guide](https://github.com/cinder/Cinder/blob/master/CONTRIBUTING.md#style-guide)). Here and below, and also on the `string ( ...` on line 11.

**Comment by dankegel** [comment]
Rebased and fixed formatting issues, I hope.

**Comment by andrewfb** [comment]
Thanks for these improvements and for the find+fix.

**Pull request merged**